gems/smithy/lib/smithy/util/hash_formatter.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Util
+    # A class to format a hash into a string.
+    class HashFormatter
+      # @option options [Boolean] :wrap (true) When `true`, the formatted hash will
+      #   be wrapped with curly braces.
+      # @option options [Boolean] :inline (false) When `true` the formatted hash will
+      #   contain no newlines.
+      # @option options [Boolean] :quote_strings (false) By default, all hash string values
+      #   must contain their own quotes. If you pass `true`, then all hash string
+      #   values will be inspected via `#inspect` which will auto-quote them.
+      def initialize(options = {})
```

Comment:
Why not use key word arguments here?

Reply:
I carried this over from v3. I actually do not prefer keyword arguments unless you need validation of required arguments. Four optional keywords in the method signature seems verbose.